core/domain/topic_fetchers_test.py

```diff
@@ -76,62 +86,85 @@ def setUp(self):
         self.signup('b@example.com', 'B')
         self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
 
-        self.user_id_a = self.get_user_id_from_email('a@example.com')
-        self.user_id_b = self.get_user_id_from_email('b@example.com')
+        self.user_id_a = self.get_user_id_from_email('a@example.com')  # type: ignore[no-untyped-call]
+        self.user_id_b = self.get_user_id_from_email('b@example.com')  # type: ignore[no-untyped-call]
         self.user_id_admin = (
-            self.get_user_id_from_email(self.CURRICULUM_ADMIN_EMAIL))
-        topic_services.update_topic_and_subtopic_pages(
+            self.get_user_id_from_email(self.CURRICULUM_ADMIN_EMAIL))  # type: ignore[no-untyped-call]
+        topic_services.update_topic_and_subtopic_pages(  # type: ignore[no-untyped-call]
             self.user_id_admin, self.TOPIC_ID, changelist, 'Added a subtopic')
 
-        self.topic = topic_fetchers.get_topic_by_id(self.TOPIC_ID)
-        self.set_curriculum_admins([self.CURRICULUM_ADMIN_USERNAME])
-        self.set_topic_managers(
-            [user_services.get_username(self.user_id_a)], self.TOPIC_ID)
-        self.user_a = user_services.get_user_actions_info(self.user_id_a)
-        self.user_b = user_services.get_user_actions_info(self.user_id_b)
-        self.user_admin = user_services.get_user_actions_info(
+        self.topic: Optional[topic_domain.Topic] = (
+            topic_fetchers.get_topic_by_id(self.TOPIC_ID)
+        )
+        self.set_curriculum_admins([self.CURRICULUM_ADMIN_USERNAME])  # type: ignore[no-untyped-call]
+        self.set_topic_managers(  # type: ignore[no-untyped-call]
+            [user_services.get_username(self.user_id_a)], self.TOPIC_ID)  # type: ignore[no-untyped-call]
+        self.user_a = user_services.get_user_actions_info(self.user_id_a)  # type: ignore[no-untyped-call]
+        self.user_b = user_services.get_user_actions_info(self.user_id_b)  # type: ignore[no-untyped-call]
+        self.user_admin = user_services.get_user_actions_info(  # type: ignore[no-untyped-call]
             self.user_id_admin)
 
-    def test_get_topic_from_model(self):
-        topic_model = topic_models.TopicModel.get(self.TOPIC_ID)
-        topic = topic_fetchers.get_topic_from_model(topic_model)
+    def test_get_topic_from_model(self) -> None:
+        topic_model: Optional[topic_models.TopicModel] = (
+            topic_models.TopicModel.get(self.TOPIC_ID)
+        )
+        # Ruling out the possibility of None for mypy type checking.
+        assert topic_model is not None
+        topic: topic_domain.Topic = (
+            topic_fetchers.get_topic_from_model(topic_model)
+        )
+        # Ruling out the possibility of None for mypy type checking.
+        assert self.topic is not None
```

Comment:
Just curious to know why is this assertion needed?
The return type of `get_topic_from_model` is Topic and it wouldn't return None right?

Reply:
Yeah, the return type of `get_topic_from_model` wouldn't be `None` but here we are asserting `self.topic`, not `topic`. Those two are different. Is that fine?

Reply:
Makes sense. My bad, I didn't notice that.